### PR TITLE
Get Agent IP every callback

### LINKED ISSUE #704: Get agent IP every callback
The imix agent right now may start before the host has an IP address as was the case for many pfsense boxes during ISTS. Since imix currently only grabs the Host IP at startup it cannot update itself once the host has an IP.

Imix should check the host's IP address every check-in this is more realistic too since hosts may change IP when using DHCP.

Currently the `primary_ip` is set in `implants/imix/src/config.rs`

```rust
impl Default for Config {
    fn default() -> Self {
        let agent = pb::c2::Agent {
            identifier: format!("imix-v{}", VERSION),
        };

        let host = pb::c2::Host {
            name: whoami::hostname(),
            identifier: get_host_id(get_host_id_path()),
            platform: get_host_platform() as i32,
            primary_ip: get_primary_ip(),
        };
```

Instead it should happen during the `callback_loop` in `implants/imix/src/agent.rs`

### PR DESCRIPTION
**Pull request by Cictrone**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

As documented in #704, we were seeing that Imix would sometimes check in faster than the networking stack would complete DHCP on some boxes. To solve this we will just refresh our `primary_ip` with each callback.

#### Which issue(s) this PR fixes:

Fixes #704

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -4,4 +4,5 @@
         "rust-analyzer"
     ],
     "rust-analyzer.check.command": "clippy",
+    "rust-analyzer.showUnlinkedFileNotification": false,
```
Comment (Cictrone) [inline .vscode/settings.json:7]: this is annoying and not very useful. I would like to leave it in but if we don't want to I'll remove it.

**Comment by Cictrone** [comment]
Some output for what it looks like while Imix waits for the box to have an IP:

```
2024-03-14T22:07:25.030Z ERROR imix::config > failed to get primary ip: Local IP address not found
 2024-03-14T22:07:25.038Z INFO  imix::agent  > claimed 0 tasks
 2024-03-14T22:07:25.038Z INFO  imix::agent  > completed callback in 0s, sleeping for 5s
 2024-03-14T22:07:30.043Z ERROR imix::config > failed to get primary ip: Local IP address not found
 2024-03-14T22:07:30.050Z INFO  imix::agent  > claimed 0 tasks
 2024-03-14T22:07:30.051Z INFO  imix::agent  > completed callback in 0s, sleeping for 5s
 2024-03-14T22:07:35.053Z ERROR imix::config > failed to get primary ip: Local IP address not found
 2024-03-14T22:07:35.064Z INFO  imix::agent  > claimed 0 tasks
 2024-03-14T22:07:35.064Z INFO  imix::agent  > completed callback in 0s, sleeping for 5s
 2024-03-14T22:07:40.069Z ERROR imix::config > failed to get primary ip: Local IP address not found
 2024-03-14T22:07:40.076Z INFO  imix::agent  > claimed 0 tasks
 2024-03-14T22:07:40.077Z INFO  imix::agent  > completed callback in 0s, sleeping for 5s
 2024-03-14T22:07:45.082Z ERROR imix::config > failed to get primary ip: Local IP address not found
 2024-03-14T22:07:45.088Z INFO  imix::agent  > claimed 0 tasks
 2024-03-14T22:07:45.089Z INFO  imix::agent  > completed callback in 0s, sleeping for 5s
 2024-03-14T22:07:50.115Z INFO  imix::agent  > claimed 0 tasks
 2024-03-14T22:07:50.115Z INFO  imix::agent  > completed callback in 0s, sleeping for 5s
 ```

**Review by hulto — APPROVED** [review]
Bless

File: .vscode/settings.json (hunk 1 above)
Comment (hulto) [reply .vscode/settings.json:7]: Idk what this does

**Pull request merged**